public/app/features/scenes/core/SceneContextObject.ts

```diff
@@ -0,0 +1,32 @@
+import { SceneObjectBase } from './SceneObjectBase';
+import { SceneTimeRange } from './SceneTimeRange';
+import { SceneObjectStatePlain, StandardSceneObjectContext } from './types';
+
+export interface SceneContextObjectState extends SceneObjectStatePlain {
+  /** Indicates this object scope uses the enclosing context */
+  inheritContext?: boolean;
+}
+
+/**
+ * A scene object that can provide data execution context
+ */
+export class SceneContextObject<TState extends SceneContextObjectState = {}> extends SceneObjectBase<TState> {
+  ctx: StandardSceneObjectContext = {
+    timeRange: new SceneTimeRange(),
+    variables: [],
+  };
+
```

Comment:
Every VizPanel will need some scoped variables, so this will mean VizPanel should extend SceneContextObject? or every scene object that needs to set a scoped variable or local query needs to be wrapped in a SceneContext?

Reply:
How would you set a local scoped variable but inherit a time range?

Reply:
Also ctx timeRange and variables are just a instance props here and not part of the SceneObject state, which means timeRange and variables live outside the scene state tree? 

This breaks the URL sync, and the whole connected tree. SceneTimeRange object is not connected to the scene root in this PR